panic: runtime error: index out of range [0] with length 0

Running this tool `go vet -vettool=$(which execinquery) ./...` on favonia/cloudflare-ddns@f227115e85a54b211ef2792f8012b0e72eaa1063 gives the following error:

```
# github.com/favonia/cloudflare-ddns/internal/api_test
panic: runtime error: index out of range [0] with length 0

goroutine 9 [running]:
github.com/lufeee/execinquery.run.func1({0x6d5d38?, 0xc000184900})
	<redacted>/go/pkg/mod/github.com/lufeee/execinquery@v1.0.0/execinquery.go:45 +0x3ae
golang.org/x/tools/go/ast/inspector.(*Inspector).Preorder(0xc0006d0258, {0x0?, 0x81aae0?, 0x0?}, 0xc0009b2cc8)
	<redacted>/go/pkg/mod/golang.org/x/tools@v0.1.9/go/ast/inspector/inspector.go:77 +0x9a
github.com/lufeee/execinquery.run(0xc00062fee0)
	<redacted>/go/pkg/mod/github.com/lufeee/execinquery@v1.0.0/execinquery.go:27 +0x77
golang.org/x/tools/go/analysis/unitchecker.run.func5.1()
	<redacted>/go/pkg/mod/golang.org/x/tools@v0.1.9/go/analysis/unitchecker/unitchecker.go:355 +0x82e
sync.(*Once).doSlow(0x0?, 0x0?)
	/usr/lib/go/src/sync/once.go:68 +0xc2
sync.(*Once).Do(...)
	/usr/lib/go/src/sync/once.go:59
golang.org/x/tools/go/analysis/unitchecker.run.func5(0x81a920?)
	<redacted>/go/pkg/mod/golang.org/x/tools@v0.1.9/go/analysis/unitchecker/unitchecker.go:307 +0x1a5
golang.org/x/tools/go/analysis/unitchecker.run.func6.1(0x0?)
	<redacted>/go/pkg/mod/golang.org/x/tools@v0.1.9/go/analysis/unitchecker/unitchecker.go:367 +0x29
created by golang.org/x/tools/go/analysis/unitchecker.run.func6
	<redacted>/go/pkg/mod/golang.org/x/tools@v0.1.9/go/analysis/unitchecker/unitchecker.go:366 +0x47
```

PS: the README seems to miss the dollar sign in `go vet -vettool=$(which execinquery) ./...`.

--- a/execinquery.go
+++ b/execinquery.go
@@ -40,6 +40,10 @@ func run(pass *analysis.Pass) (interface{}, error) {
 				return
 			}
 
+			if "database/sql" != pass.TypesInfo.Uses[selector.Sel].Pkg().Path() {
+				return
+			}
+
 			if !strings.Contains(selector.Sel.Name, "Query") {
 				return
 			}

--- a/execinquery.go
+++ b/execinquery.go
@@ -22,18 +22,26 @@ var Analyzer = &analysis.Analyzer{
 }
 
 func run(pass *analysis.Pass) (interface{}, error) {
-	inspect := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
+	result := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
 
-	inspect.Preorder(nil, func(n ast.Node) {
+	nodeFilter := []ast.Node{
+		(*ast.CallExpr)(nil),
+	}
+
+	result.Preorder(nodeFilter, func(n ast.Node) {
 		switch n := n.(type) {
 		case *ast.CallExpr:
+			if len(n.Args) < 1 {
+				return
+			}
+
 			selector, ok := n.Fun.(*ast.SelectorExpr)
 			if !ok {
-				break
+				return
 			}
 
 			if !strings.Contains(selector.Sel.Name, "Query") {
-				break
+				return
 			}
 
 			var i int
@@ -45,28 +53,35 @@ func run(pass *analysis.Pass) (interface{}, error) {
 			switch arg := n.Args[i].(type) {
 			case *ast.BasicLit:
 				s = strings.Replace(arg.Value, "\"", "", -1)
+
 			case *ast.Ident:
 				stmt, ok := arg.Obj.Decl.(*ast.AssignStmt)
 				if !ok {
-					break
+					return
 				}
+
 				for _, stmt := range stmt.Rhs {
 					basicLit, ok := stmt.(*ast.BasicLit)
 					if !ok {
 						continue
 					}
+
 					s = strings.Replace(basicLit.Value, "\"", "", -1)
 				}
+
 			default:
-				break
+				return
 			}
 
 			if strings.HasPrefix(strings.ToLower(s), "select") {
-				break
+				return
 			}
-			s = strings.ToTitle(strings.Split(s, " ")[0])
+
+			s = strings.ToTitle(strings.SplitN(s, " ", 2)[0])
+
 			pass.Reportf(n.Fun.Pos(), "It's better to use Execute method instead of %s method to execute `%s` query", selector.Sel.Name, s)
 		}
 	})
+
 	return nil, nil
 }

